web/modules/custom/genres_to_terms/genres_to_terms.module

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @file
+ * Creates taxonomy terms based on the Genres field's input.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Contains hooks and callbacks for genres_to_terms.module.
+ */
+function genres_to_terms_entity_presave(EntityInterface $entity): void {
+  if ($entity->bundle() === 'product') {
+    unset($entity->field_genre_tags);
```

Comment:
Why?

Reply:
Otherwise if i modify the genre field after creating it(for example replace "Shooter" with Open-World) with,the new taxonomy terms will be added on top of the existing ones,this was the only way i managed to avoid that.

Reply:
It is not needed if you overwrite the field's value at the end with the collected term ids.

Reply:
You"re right,i forgot that the code uses "set" now instead of simply just adding them to it!